xarray/core/dataset.py

```diff
@@ -2179,5 +2179,77 @@ def real(self):
     def imag(self):
         return self._unary_op(lambda x: x.imag, keep_attrs=True)(self)
 
+    def get_variables_by_attributes(self, **kwargs):
+        """Returns a ``Dataset`` with variables that match specific conditions.
+        Can pass in ``key=value ``or ``key=callable``.  Variables are returned
+        that contain all of the matches or callable returns True.  If using a
+        callable note that it should accept a single parameter only,
+        the attribute value.
+
+        Parameters
+        ----------
+        **kwargs : key=value
+            key : str
+                Filter variables based on a string attribute.
+            value : str or callable
+                Filter variables based on callable False/True answer.
+
+        Returns
+        -------
+        new : Dataset
+
```

Comment:
I'll suggest a few more modifications:

```Python
    """
    ...
        Parameters
        ----------
        **kwargs : key=value
            key : str
                Attribute name.
            value : callable or obj
                If value is a callable, it should return a boolean in the form of bool = func(attr) where attr is da.attrs[key].  
                Otherwise, value will be compared to the each DataArray's attrs[key].

        Returns
        -------
        new : Dataset
            New dataset with variables filtered by attribute.

    ...
    """
```

I think the `value` section could use a bit more thought.